### ISSUE TITLE
How to access middleware context in CROW_ROUTE

### ISSUE THREAD

**Issue by biot023** [ipkn/crow#240]
Hello -- I have a bit of middleware that stores an object (a struct representing a user) in the middleware's context.
I now need to be able to access that user object from the context for this request in a CROW_ROUTE.
I've tried something like this:

    CROW_ROUTE(...) {
        const UserModel & user = app.get_middleware<MyMiddleware<UserModel>>()::context.get_user();
    }

But obviously that doesn't work, and even if it did it wouldn't be the context specific to the current request that I need.
Could anyone advise me as to how to access the middleware context for the current request?
Thanks for any and all assistance,
   Doug.

**Comment by biot023** [comment]
Ah. My question rather assumes that one can access the middlewares' contexts in a CROW_ROUTE. As did I.
But as I continue to browse the source, I'm seeing less and less hope that that is the case.
What a dumb assumption I've started out with!
If someone could categorically tell me that it is or isn't possible here, that would be great.
I'm getting close to thinking that it isn't possible in the meantime...
Cheers,
   Doug.

**Comment by ipkn** [comment]
Sorry for late reply.

It is possible to access to any middleware context from the handler.

from MW - `all_ctx.get<A>()`
from handler - `app.get_context<A>(req)`

`app` knows installed middlewares, so the API have to be attached to `app`.

One of problem of crow is lack of documentation. I fill it soon after I got some free time.

**Comment by biot023** [comment]
Ah, that's really interesting, thank you!
I worked around it for now, but I think I'll engineer back to using something like your example using get_context() -- thanks so much again!

**Comment by biot023** [comment]
Yeah, that's miles better, and would've been a bit more intuitive to me were my brain a better model -- thank you ever so much for your help with that, @ipkn !

**Comment by qimj** [comment]
@biot023 you mean you are working for something like coroutine with crow? did that work? i'm interrested.

**Comment by biot023** [comment]
Hi @qimj -- I'm not sorry what you mean, sorry.
I was struggling to get the context within a route and so came up with a rather inelegant hack where I was storing what I needed (just strings, in this case) in a custom header on the request.
However @ipkn got back to me and showed me the proper way to get at the context I needed, so I changed my code to use that and it was miles better.
Is that what you're asking?
It's been a while since I've been able to work on this project but I can refresh my memory of it (when I can grab a minute away from work) if it might help.
Cheers,
   Doug.
